src/varity/vcf_to_hgvs/protein.clj

```diff
@@ -175,10 +175,12 @@
                                            :extension
                                            :frame-shift)
               (and (pos? nprefo) (= (first palt-only) \*)) :substitution
-              (not= ref-prot-rest alt-prot-rest) (if (or (and (empty? palt-only)
-                                                              (= (first alt-prot-rest) \*))
-                                                         (= (last palt-only) \*)) :fs-ter-substitution
-                                                     :frame-shift)
+              (not= ref-prot-rest alt-prot-rest) (if (or (and (= (first alt-prot-rest) \*)
+                                                              (>= nprefo npalto)
+                                                              (= palt (subs pref 0 (count palt))))
+                                                         (= (last palt-only) \*))
+                                                   :fs-ter-substitution
+                                                   :frame-shift)
```

Comment:
The `fs-ter-subsitution` detection added by #15 is insufficient because `varity.vcf-to-hgvs.common/diff-bases` returns right-side commons as well as left-side commons. In this case,

```clojure
(require '[varity.vcf-to-hgvs.common :refer [diff-bases]])

(diff-bases "SGI" "I")
;;=> ["SG" "" 0 1] ; [pref-only palt-only offset _]
```

So `(empty? palt-only)` cannot detect frameshift in the changed proteins and wrongly detect it as `fs-ter-substitution`. A direct comparison between `pref` and `palt` is needed.

Reply:
Does that mean `:fs-ter-substitution` had been correct for `(diff-bases "SGI" "S")` but not for `(diff-bases "SGI" "I")`?

Reply:
Yes. If the ref proteins `...SGIXXXX...` change into the alt proteins `...S*YYYY...`, that is substitution of `G` and `*`, not frameshift.

In #58 case, the ref proteins `...SGIXXXX...` change into the alt proteins `...I*YYYY...`. That is frameshift caused by the change of `SGI` to `I`.

Reply:
Thanks. That makes sense!